liquidprompt

```diff
@@ -2341,37 +2345,69 @@ _lp_wifi() {
 # Get the battery status in percent.
 case "$LP_OS" in
     Linux)
-    _lp_battery() {
-        (( LP_ENABLE_BATT )) || return 5
+    __lp_battery_sysfs() {
+        local power_supply
+        for power_supply in "${_LP_LINUX_POWERSUPPLY_PATH}/"*; do
+            if [[ -r "${power_supply}/type" && $(< "${power_supply}/type") == 'Battery' && \
+                -r "${power_supply}/present" && $(< "${power_supply}/present") == '1' && \
+                -r "${power_supply}/status" && -r "${power_supply}/capacity" && \
+                (! -f "${power_supply}/scope" || \
+                (-f "${power_supply}/scope" && $(< "${power_supply}/scope") == 'System')) ]]; then
```

Comment:
You replaced the `$(< file)` syntax below, but there are all of them still inside this if check. This will slow things down, so we need to rewrite this somehow. Maybe like:
```bash
  if ! [[ -r "${power_supply}/type" && -r "${power_supply}/present" && \
      -r "${power_supply}/status" && -r "${power_supply}/capacity" ]]; then
    return 1
  fi

  local power_supply_type power_supply_present power_supply_scope
  IFS= read -r power_supply_type <"${power_supply}/type"
  [[ $power_supply_type == 'Battery' ]] || return 1
  IFS= read -r power_supply_present <"${power_supply}/present"
  [[ $power_supply_present == '1' ]] || return 1
  IFS= read -r power_supply_present <"${power_supply}/present"
  [[ $power_supply_present == '1' ]] || return 1

  # Scope is a property of a power supply
  # Scope = System or missing - power supply powers the system
  # Scope = Device - power supply powers a device
  if [[ -f "${power_supply}/scope" ]]; then
    IFS= read -r power_supply_scope <"${power_supply}/scope"
    [[ $power_supply_scope == 'System' ]] || return 1
  fi

  IFS= read -r lp_battery_status <"${power_supply}/status"
  IFS= read -r lp_battery <"${power_supply}/capacity"
  return 0
```